src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -742,35 +743,163 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
+        private ClaimsPrincipal ValidateToken(string token, JwtSecurityToken outerToken, TokenValidationParameters validationParameters, out SecurityToken signatureValidatedToken)
+        {
+            BaseConfiguration currentConfiguration = null;
+            if (validationParameters.ConfigurationManager != null)
+            {
+                try
+                {
+                    currentConfiguration = validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
+                }
+                catch (Exception ex)
+                {
+                    // The exception is not re-thrown as the TokenValidationParameters may have the issuer and signing key set
+                    // directly on them, allowing the library to continue with token validation.
+                    LogHelper.LogWarning(LogHelper.FormatInvariant(TokenLogMessages.IDX10261, validationParameters.ConfigurationManager.MetadataAddress, ex.ToString()));
+                }
+            }
+
+            Exception exceptionThrown;
+            ClaimsPrincipal claimsPrincipal = outerToken != null ? ValidateJWE(token, outerToken, validationParameters, currentConfiguration, out signatureValidatedToken, out exceptionThrown) :
+                ValidateJWS(token, validationParameters, currentConfiguration, out signatureValidatedToken, out exceptionThrown);
+            if (validationParameters.ConfigurationManager != null)
+            {
+                if (claimsPrincipal != null)
+                {
+                    // Set current configuration as LKG if it exists and is not the same as the LKG.
+                    if (currentConfiguration != null && currentConfiguration != validationParameters.ConfigurationManager.LastKnownGoodConfiguration)
+                        validationParameters.ConfigurationManager.LastKnownGoodConfiguration = currentConfiguration;
+
+                    return claimsPrincipal;
+                }
+                // using 'GetType()' instead of 'is' as SecurityTokenUnableToValidException (and others) extend SecurityTokenInvalidSignatureException
+                // we want to make sure that the clause for SecurityTokenUnableToValidateException is hit so that the ValidationFailure is checked
+                else if (exceptionThrown.GetType().Equals(typeof(SecurityTokenInvalidSignatureException))
+                   || exceptionThrown is SecurityTokenInvalidSigningKeyException
+                   || exceptionThrown is SecurityTokenInvalidIssuerException
+                   || (exceptionThrown is SecurityTokenUnableToValidateException
+                   // we should not try to revalidate with the LKG or request a refresh if the token has an invalid lifetime
+                   && (exceptionThrown as SecurityTokenUnableToValidateException).ValidationFailure != ValidationFailure.InvalidLifetime)
```

Comment:
consider:

(exceptionThrown as SecurityTokenUnableToValidateException)?.ValidationFailure != ValidationFailure.InvalidLifetime

should have the same behavior but less verbose